nemo/nemo/utils/decorators/deprecated.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) tkornuta, NVIDIA AI Applications Team. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__author__ = "Tomasz Kornuta"
+
+import nemo
+
+
+class deprecated(object):
+    """ Decorator class used for indicating that a function is
+    deprecated and going to be removed.
+    Tracks down which functions printed the warning and
+    will print it only once per function.
+    """
+
+    # Static variable - list of names of functions that we already printed
+    # the warning for.
+    warned_functions = []
+
+    def __init__(self, version=None, alternative_function=None):
+        """
+        If there are decorator arguments, the function
+        to be decorated is not passed to the constructor!
+        """
+        self.version = version
+        self.alternative_function = alternative_function
+
+    def __call__(self, func):
+        """
+        If there are decorator arguments, __call__() is only called
+        once, as part of the decoration process! You can only give
+        it a single argument, which is the function object.
+        """
+
+        def wrapper(*args, **kwargs):
+
+            # Check if we already warned about that function.
+            if func.__name__ not in deprecated.warned_functions:
```

Comment:
This is nice, but if we just use the warning.warn library. The user can set all warnings to only print once. Like so:
```
import warnings
warnings.filterwarnings("ignore", category=UserWarning)
```
I feel like this would be a nicer way of handling it, rather than us adding and maintaining a list.

Reply:
hmmm, interesting, I didn't know that one. How is it working? Filtering by the actual "message"?

Reply:
also, I am not sure if we want this warning to act exactly as the other ones... the only and desired behavior is to print the warning once, not "polluting" the log too much, right?

so maybe let's stay with this for now, and we will update it when we will need to, ok?

Reply:
According to https://docs.python.org/3/howto/logging.html :
```
warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning

logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted
```
I would argue that warnings.warn() is more appropriate than logging.warning for deprecation messages. We can probably talk about this with the team though.